From aa8bacf821e5c8ae6dd8cae8d64011c741659945 Mon Sep 17 00:00:00 2001
From: Leo Wen <leo.wen@rock-chips.com>
Date: Fri, 21 Sep 2018 16:39:09 +0800
Subject: [PATCH] arm64: dts: rk3399-sapphire-excavator-linux: support
 isp1&ov13850 node

Delete cif_isp10 and all cif_isp10_cameras node

Change-Id: I6f5bf3ae7001fcb456a02abbb3ef327ffedc3b76
Signed-off-by: Leo Wen <leo.wen@rock-chips.com>
---
 .../rk3399-sapphire-excavator-linux.dts       | 254 +++++++-----------
 1 file changed, 104 insertions(+), 150 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts
index 6b5e3219407f..6bf488add069 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts
@@ -141,19 +141,101 @@
 	};
 };
 
-&cif_isp0 {
-	rockchip,camera-modules-attached =
-		<&camera0 &camera1 &camera2 &camera3>;
+&rkisp1_0 {
 	status = "okay";
+
+	port {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		isp0_mipi_in: endpoint@0 {
+			reg = <0>;
+			remote-endpoint = <&dphy_rx0_out>;
+		};
+	};
+};
+
+&mipi_dphy_rx0 {
+	status = "okay";
+
+	ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		port@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			mipi_in_ucam0: endpoint@1 {
+				reg = <1>;
+				remote-endpoint = <&ucam_out0>;
+				data-lanes = <1 2>;
+			};
+		};
+
+		port@1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			dphy_rx0_out: endpoint@0 {
+				reg = <0>;
+				remote-endpoint = <&isp0_mipi_in>;
+			};
+		};
+	};
 };
 
 &isp0_mmu {
 	status = "okay";
 };
 
-&cif_isp1 {
-	rockchip,camera-modules-attached = <&camera4>;
+&rkisp1_1 {
 	status = "okay";
+
+	port {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		isp1_mipi_in: endpoint@0 {
+			reg = <0>;
+			remote-endpoint = <&dphy_tx1rx1_out>;
+		};
+	};
+};
+
+&mipi_dphy_tx1rx1 {
+	status = "okay";
+
+	ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		port@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			mipi_in_ucam1: endpoint@1 {
+				reg = <1>;
+				/* Unlinked camera */
+				//remote-endpoint = <&ucam_out1>;
+				data-lanes = <1 2>;
+			};
+		};
+
+		port@1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			dphy_tx1rx1_out: endpoint@0 {
+				reg = <0>;
+				remote-endpoint = <&isp1_mipi_in>;
+			};
+		};
+	};
 };
 
 &isp1_mmu {
@@ -227,158 +309,30 @@
 		rst_gpio_number = <&gpio4 22 GPIO_ACTIVE_HIGH>;
 	};
 
-	camera0: camera-module@60 {
-		status = "okay";
-		compatible = "ovti,ov7750-v4l2-i2c-subdev";
-		reg = < 0x60 >;
-		device_type = "v4l2-i2c-subdev";
-
-		clocks = <&cru SCLK_CIF_OUT>;
-		clock-names = "clk_cif_out";
-
-		pinctrl-names = "rockchip,camera_default",
-			"rockchip,camera_sleep";
-		pinctrl-0 = <&cam0_default_pins>;
-		pinctrl-1 = <&cam0_sleep_pins>;
-
-		rockchip,pwr-gpio = <&gpio2 10 GPIO_ACTIVE_HIGH>;
-		rockchip,camera-module-mclk-name = "clk_cif_out";
-		rockchip,camera-module-facing = "back";
-		rockchip,camera-module-name = "cmk-cb0695-fv1";
-		rockchip,camera-module-len-name = "lg9569a2";
-		rockchip,camera-module-fov-h = "133.0";
-		rockchip,camera-module-fov-v = "100.1";
-		rockchip,camera-module-orientation = <0>;
-		rockchip,camera-module-iq-flip = <0>;
-		rockchip,camera-module-iq-mirror = <0>;
-		rockchip,camera-module-flip = <0>;
-		rockchip,camera-module-mirror = <0>;
-
-		rockchip,camera-module-defrect0 = <640 480 0 0 640 480>;
-		rockchip,camera-module-defrect1 = <640 480 0 0 640 480>;
-		rockchip,camera-module-defrect2 = <640 480 0 0 640 480>;
-		rockchip,camera-module-defrect3 = <640 480 0 0 640 480>;
-		rockchip,camera-module-flash-support = <0>;
-		rockchip,camera-module-mipi-dphy-index = <0>;
-	};
-
-	camera1: camera-module@1 {
+	ov13850: ov13850@10 {
+		compatible = "ovti,ov13850";
 		status = "okay";
-		compatible = "toshiba,tc358749xbg-v4l2-i2c-subdev";
-		reg = < 0x0f >;
-		device_type = "v4l2-i2c-subdev";
-
-		clocks = <&cru SCLK_CIF_OUT>;
-		clock-names = "clk_cif_out";
-
-		pinctrl-names = "default";
-		pinctrl-0 = <&hdmiin_gpios>;
-
-		//rockchip,pwr-gpio = <&gpio2 6 GPIO_ACTIVE_HIGH>;
-		//rckchip,pwr-snd-gpio = <&gpio2 9 GPIO_ACTIVE_HIGH>;
-		//rockchip,pwr-trd-gpio = <&gpio2 5 GPIO_ACTIVE_HIGH>;
-		//rockchip,rst-gpio = <&gpio2 7 GPIO_ACTIVE_HIGH>;
-		power-gpios = <&gpio2 6 GPIO_ACTIVE_HIGH>;
-		power18-gpios = <&gpio2 9 GPIO_ACTIVE_HIGH>;
-		power33-gpios = <&gpio2 5 GPIO_ACTIVE_HIGH>;
-		csi-ctl-gpios = <&gpio2 10 GPIO_ACTIVE_HIGH>;
-		stanby-gpios = <&gpio2 8 GPIO_ACTIVE_HIGH>;
-		reset-gpios = <&gpio2 7 GPIO_ACTIVE_HIGH>;
-		int-gpios = <&gpio2 12 GPIO_ACTIVE_HIGH>;
-
-		rockchip,camera-module-mclk-name = "clk_cif_out";
-		rockchip,camera-module-mipi-dphy-index = <0>;
-	};
-
-	camera2: camera-module@10 {
-		status = "okay";
-		compatible = "omnivision,ov13850-v4l2-i2c-subdev";
-		reg = < 0x10 >;
-		device_type = "v4l2-i2c-subdev";
-		clocks = <&cru SCLK_CIF_OUT>;
-		clock-names = "clk_cif_out";
-		pinctrl-names = "rockchip,camera_default",
-				"rockchip,camera_sleep";
-		pinctrl-0 = <&cam0_default_pins>;
-		pinctrl-1 = <&cam0_sleep_pins>;
-		rockchip,pd-gpio = <&gpio1 4 GPIO_ACTIVE_LOW>;
-		//rockchip,pwr-gpio = <&gpio2 10 GPIO_ACTIVE_HIGH>;
-		/* mipi switch control */
-		rockchip,rst-gpio = <&gpio2 10 GPIO_ACTIVE_LOW>;
-		rockchip,camera-module-mclk-name = "clk_cif_out";
-		rockchip,camera-module-facing = "back";
-		rockchip,camera-module-name = "cmk-cb0695-fv1";
-		rockchip,camera-module-len-name = "lg9569a2";
-		rockchip,camera-module-fov-h = "66.0";
-		rockchip,camera-module-fov-v = "50.1";
-		rockchip,camera-module-orientation = <0>;
-		rockchip,camera-module-iq-flip = <0>;
-		rockchip,camera-module-iq-mirror = <0>;
-		rockchip,camera-module-flip = <1>;
-		rockchip,camera-module-mirror = <0>;
-
-		rockchip,camera-module-defrect0 = <2112 1568 0 0 2112 1568>;
-		rockchip,camera-module-defrect1 = <4224 3136 0 0 4224 3136>;
-		rockchip,camera-module-defrect3 = <3264 2448 0 0 3264 2448>;
-		rockchip,camera-module-flash-support = <1>;
-		rockchip,camera-module-mipi-dphy-index = <0>;
-	};
-
-	camera3: camera-module@36 {
-		status = "okay";
-		compatible = "omnivision,ov4689-v4l2-i2c-subdev";
-		reg = <0x36>;
-		device_type = "v4l2-i2c-subdev";
+		reg = <0x10>;
 		clocks = <&cru SCLK_CIF_OUT>;
-		clock-names = "clk_cif_out";
-		pinctrl-names = "rockchip,camera_default",
-				"rockchip,camera_sleep";
-		pinctrl-0 = <&cam0_default_pins>;
-		pinctrl-1 = <&cam0_sleep_pins>;
-		rockchip,pd-gpio = <&gpio1 4 GPIO_ACTIVE_LOW>;
-		//rockchip,pwr-gpio = <&gpio3 13 0>;
-		/* mipi switch control*/
-		rockchip,rst-gpio = <&gpio2 10 GPIO_ACTIVE_LOW>;
-		rockchip,camera-module-mclk-name = "clk_cif_out";
-		rockchip,camera-module-facing = "back";
-		rockchip,camera-module-name = "LA6111PA";
-		rockchip,camera-module-len-name = "YM6011P";
-		rockchip,camera-module-fov-h = "116";
-		rockchip,camera-module-fov-v = "61";
-		rockchip,camera-module-orientation = <0>;
-		rockchip,camera-module-iq-flip = <0>;
-		rockchip,camera-module-iq-mirror = <0>;
-		rockchip,camera-module-flip = <0>;
-		rockchip,camera-module-mirror = <1>;
-
-		rockchip,camera-module-defrect0 = <2688 1520 0 0 2688 1520>;
-		rockchip,camera-module-flash-support = <0>;
-		rockchip,camera-module-mipi-dphy-index = <0>;
+		clock-names = "xvclk";
+
+		/* conflict with csi-ctl-gpios */
+		reset-gpios = <&gpio2 10 GPIO_ACTIVE_HIGH>;
+		pwdn-gpios = <&gpio1 4 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "rockchip,camera_default";
+		pinctrl-0 = <&cif_clkout>;
+
+		port {
+			ucam_out0: endpoint {
+				remote-endpoint = <&mipi_in_ucam0>;
+				data-lanes = <1 2>;
+			};
+		};
 	};
 };
 
 &i2c4 {
 	status = "okay";
-
-	camera4: camera-module@2 {
-		status = "okay";
-		compatible = "adi,adv7181-v4l2-i2c-subdev";
-		reg = < 0x21 >;
-		device_type = "v4l2-i2c-subdev";
-
-		clocks = <&cru SCLK_CIF_OUT>;
-		clock-names = "clk_cif_out";
-
-		pinctrl-names = "rockchip,camera_default",
-			"rockchip,camera_sleep";
-		pinctrl-0 = <&isp_dvp_d0d7>;
-		pinctrl-1 = <&cam0_sleep_pins>;
-
-		rockchip,camera-module-defrect0 = <720 480 0 13 720 480>;
-		rockchip,camera-module-defrect1 = <720 576 0 0 720 576>;
-		rockchip,camera-module-mclk-name = "clk_cif_out";
-		rockchip,camera-module-mipi-dphy-index = <1>;
-	};
 };
 
 &pcie_phy {
-- 
2.35.3

